core/src/transport/memory.rs

```diff
@@ -29,19 +29,50 @@ use rw_stream_sink::RwStreamSink;
 use std::{collections::hash_map::Entry, error, fmt, io, num::NonZeroU64};
 
 lazy_static! {
-    static ref HUB: Mutex<FnvHashMap<NonZeroU64, mpsc::UnboundedSender<Channel<Bytes>>>> = Mutex::new(FnvHashMap::default());
+    static ref HUB: Mutex<FnvHashMap<NonZeroU64, mpsc::Sender<Channel<Bytes>>>> = Mutex::new(FnvHashMap::default());
 }
 
+static CHAN_BUF_SIZE: usize = 1000;
+
 /// Transport that supports `/memory/N` multiaddresses.
 #[derive(Debug, Copy, Clone, Default)]
 pub struct MemoryTransport;
 
+pub struct DialFuture {
```

Comment:
Needs a small doc-comment. Nothing big, just *something*.
```suggestion
/// Connection to a `MemoryTransport` currently being opened.
pub struct DialFuture {
```